waspc/data/Generator/templates/server/src/auth/providers/email/utils.ts

```diff
@@ -2,80 +2,68 @@
 import { sign } from '../../../core/auth.js'
 import { emailSender } from '../../../email/index.js';
 import { Email } from '../../../email/core/types.js';
-import { rethrowPossiblePrismaError } from '../../utils.js'
-import prisma from '../../../dbClient.js'
+import {
+  rethrowPossiblePrismaError,
+  updateAuthIdentityProviderData,
+  findAuthIdentity,
+  deserializeProviderData,
+} from '../../utils.js';
 import waspServerConfig from '../../../config.js';
-import { type {= userEntityUpper =} } from '../../../entities/index.js'
+import { type {= userEntityUpper =}, type {= authEntityUpper =} } from '../../../entities/index.js'
 
+type {= authEntityUpper =}Id = {= authEntityUpper =}['id']
 type {= userEntityUpper =}Id = {= userEntityUpper =}['id']
 
-export async function updateUserEmailVerification(userId: {= userEntityUpper =}Id): Promise<void> {
-  try {
-    await prisma.{= userEntityLower =}.update({
-      where: { id: userId },
-      data: { isEmailVerified: true },
-    })
-  } catch (e) {
-    rethrowPossiblePrismaError(e);
-  }
+type AuthWithId = {
+  id: {= authEntityUpper =}Id,
```

Comment:
I wanted to keep the source of the trust for the entity names and fields in Haskell.

Reply:
Makes sense although I wonder if in this case it maybe went a bit too far. THere is som line, because we could also keep stuff like file names and all variables names in Haskell but we don't. If you like this, ok, I probably wouldn't keep this in Haskell though.

Reply:
Ok, after I read the whole PR, I get it, it makes sense.